### DISCUSSION TITLE
Templates in Visual Studio 2022

### DISCUSSION THREAD

**Discussion by tompinnacle** [unoplatform/uno#13345] [General]
Yesterday I downloaded all the latest parts to use Uno for the first time and have basically got quite confused.

I haven't found any examples/tutorials that use the solution template from withing Visual Studio, and it seems the solution generates by the CLI is different to that from withing VS.

The first difference I notice is the folder called Backend - no idea what this is meant for?

Then there is the .DataContracts section - again no idea what this is meant for, but I do note that it has the WeatherForecast example in there - is this a documented and explained example somewhere?

The rest of the solution looks pretty similar to the more recent example I can find online.  Other than Shared seems to replaced with a tree node that is the name of your project?

However because of the 2 difference above I don't know where models and view models should be for a MVVM solution - there is a Models folder under Business folder.

Is the VS solution template up to date and correct or the CLI more up to date?  Is there a guide for the VS template?

**Comment by LuisAlfredo92** [comment]
I'll say what I know but I'm not 100% sure about some things:

> The first difference I notice is the folder called Backend - no idea what this is meant for?

I guess that's the folder that contains the backend for Web applications like Blazor server app, but it might be autogenerated code

> Then there is the .DataContracts section

That `DataContracts` project is to separate your data requests logic (like API fetches or data bases queries) to make a good modular project, and it should be asynchronous to use the `IFeed` container (suggested)

> I don't know where models and view models should be for a MVVM solution

They should be on the shared project, there you can create your own folder tree of your files, but if you're using the default navigation, you have to register them on your `App.cs file`, at the bottom on the `RegisterRoutes` function

**Reply by tompinnacle** [reply]
@LuisAlfredo92 

Thanks for the reply.

Everything you say makes sense.  I found the backend is one of the options when you customise the project at the beginning.

I think the big issue here it the guides seem to be out of date/incomplete on the current functionality.  You then start to Google and YouTube and most things you find are 1+ years old, there is very little that is current an even less that us using the templates to generate the solution/project

**Reply by digger73** [reply]
A little late here, but I fully agree. Making it rather difficult to do serious development using UNO.

**Reply by francoistanguay** [reply]
Have you used the latest Wizard and selected only what you needed? Can you please share what might still be confusing?
